pkg/cmd/server/server.go

```diff
@@ -832,3 +888,34 @@ func (s *server) runProfiler() {
 		s.logger.WithError(errors.WithStack(err)).Error("error running profiler http server")
 	}
 }
+
+// CSIInformerFactoryWrapper is a proxy around the CSI SharedInformerFactory that checks the CSI feature flag before performing operations.
+type CSIInformerFactoryWrapper struct {
+	factory snapshotvebeta1informers.SharedInformerFactory
+}
+
+func NewCSIInformerFactoryWrapper(c snapshotvebeta1client.Interface) *CSIInformerFactoryWrapper {
+	// If no namespace is specified, all namespaces are watched.
+	// This is desirable for VolumeSnapshots, as we want to query for all VolumeSnapshots across all namespaces using this informer
+	w := &CSIInformerFactoryWrapper{}
+
+	if features.IsEnabled("EnableCSI") {
+		w.factory = snapshotvebeta1informers.NewSharedInformerFactoryWithOptions(c, 0)
+	}
+	return w
+}
+
+// Start proxies the Start call to the CSI SharedInformerFactory.
+func (w *CSIInformerFactoryWrapper) Start(stopCh <-chan struct{}) {
+	if features.IsEnabled("EnableCSI") {
+		w.factory.Start(stopCh)
+	}
+}
+
+// WaitForCacheSync proxies the WaitForCacheSync call to the CSI SharedInformerFactory.
+func (w *CSIInformerFactoryWrapper) WaitForCacheSync(stopCh <-chan struct{}) map[reflect.Type]bool {
+	if features.IsEnabled("EnableCSI") {
+		return w.factory.WaitForCacheSync(stopCh)
+	}
+	return make(map[reflect.Type]bool, 0)
```

Comment:
I think you could just return nil here, don't need to make an empty map

Reply:
I actually don't think I can - iirc, I returned `nil` initially, and it broke. I'll test again, though.

Reply:
This appears to work, so I was thinking of some other error with a nil map. I'll include this change in my next set of review feedback changes.